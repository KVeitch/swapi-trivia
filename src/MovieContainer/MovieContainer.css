.movie__container { 
    background-color: rgba(67, 52, 45, 0.7);
    color: white;
    display: flex;
    height: 53vh;
    margin-left: 7%;
    width: 85%;
    overflow:scroll;
}