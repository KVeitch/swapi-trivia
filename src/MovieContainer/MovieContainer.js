import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import PropTypes from 'prop-types';
import './MovieContainer.css';
import loading from '../images/loading.gif';

const MovieContainer = props => {
  if(props.movies.length === 0){
    return(
      <>
      <h1>Loading..</h1>
      <div className='div__image'>
        <p className='loading__text'>Please be patient,</p>
        <p className='loading__text'>the service droids are doing their best.</p>
        <img className='loading__image' src={loading} alt="BB-8 droid rolling back and forth" />
      </div>
      </>
    )
  }

  const movieMap = [4,5,6,1,2,3,7]
  const movieList = props.movies.map((movie, i) => 
      <MovieCard 
        title={movie.title}
        episode={movie.episode_id}
        movieImage={movie.movieImage}
        key={movie.episode_id}
        releaseYear={movie.release_date}
        characters={movie.characters}
        changeSelectedMovie={props.changeSelectedMovie}
        setCurrentCharacters={props.setCurrentCharacters}
        id={movieMap[i]}
        />)

  return (
    <section className='movie__container'>
      {movieList}
    </section>
  )
}

export default MovieContainer;

MovieContainer.propTypes ={
  movies:PropTypes.array,
  setCurrentCharacters:PropTypes.func,
  changeSelectedMovie:PropTypes.func, 
  characters:PropTypes.array
}