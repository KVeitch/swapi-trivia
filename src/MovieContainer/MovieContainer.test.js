import React from 'react';
import { shallow } from 'enzyme';
import MovieContainer from '../MovieContainer/MovieContainer';

describe('Movie Container', () => {

    it('should match the snapshot with all the data passed in correctly', () => {
        const wrapper =
            shallow(<MovieContainer
                    movies={[]}
                    changeSelectedMovie={jest.fn()}/>)
        expect(wrapper).toMatchSnapshot();
    });
});