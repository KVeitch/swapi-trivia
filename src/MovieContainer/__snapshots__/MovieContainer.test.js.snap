// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Container should match the snapshot with all the data passed in correctly 1`] = `
<Fragment>
  <h1>
    Loading..
  </h1>
  <div
    className="div__image"
  >
    <p
      className="loading__text"
    >
      Please be patient,
    </p>
    <p
      className="loading__text"
    >
      the service droids are doing their best.
    </p>
    <img
      alt="BB-8 droid rolling back and forth"
      className="loading__image"
      src="loading.gif"
    />
  </div>
</Fragment>
`;
